nilearn/_utils/numpy_conversions.py

```diff
@@ -9,22 +9,15 @@
 
 
 def _asarray(arr, dtype=None, order=None):
-    # np.asarray does not take "K" and "A" orders in version 1.3.0
-    if order in ("K", "A", None):
-        if (arr.itemsize == 1 and dtype in (bool, np.bool_)) or (
-            arr.dtype in (bool, np.bool_) and np.dtype(dtype).itemsize == 1
-        ):
-            ret = arr.view(dtype=dtype)
-        else:
-            ret = np.asarray(arr, dtype=dtype)
-    elif (
+    if (
         (arr.itemsize == 1 and dtype in (bool, np.bool_))
         or (arr.dtype in (bool, np.bool_) and np.dtype(dtype).itemsize == 1)
     ) and (
         order == "F"
         and arr.flags["F_CONTIGUOUS"]
         or order == "C"
         and arr.flags["C_CONTIGUOUS"]
+        or order in ("K", "A", None)
```

Comment:
nope
not from the top of my head
would need to look into numpy

Reply:
I guess, it is for performance concerns. As far as I understand, this is a case where numpy does not need to copy the specified `arr` itself, so if a copy is not required, it sends a view.

Reply:
As far as I understand, this is also for performance issues. 

`np.asarray` does not create a copy of array unless it is necessary. When `dtype` or `order` changes it creates a copy. 

I think this is to avoid unnecessary copying of array when the the size of the `dtype` speficied is compatible with the `dype` of array and the `order` does not change. This is done only when either `dtype` specified or `dtype` of the array is `bool`. When the condition is satisfied, it returns a view instead of copying the underlying array.